docs: plate reader manual

jwwojak opened the pull request:
# Overview

This PR migrates the plate reader instruction manual to MkDocs.

RTC-783

## Test Plan and Hands on Testing

Check for grammar, style, and closely match the original source.

## Changelog

### Manual reog (minor)

Some sections of the original PDF were very short and looked weird as standalone pages. These have been moved into the "Product Specifications" section.  Specifically, these:

- Measurement specs
- Detection wavelengths
- Status light
- Input/output (the power specs)

### New CSS

Added  to `shared/instruction-manual.css`. This creates a status dot that shows a 🌈 or color spectrum for one of the dots. The reader LED has a status that shows multiple colors.

```
/* Creates dot with red to blue spectrum lines. See plate reader LED table. */

.status-dot.rainbow {
  background-image: linear-gradient(to right, red, orange, yellow, green, blue);
  background-color: transparent; 
}
```

Code called in LED status light table:

```
<tr>
      <td><span class="status-dot-container"><span class="status-dot rainbow"></span>Various</span></td>
      <td>Solid</td>
      <td>Initialization/measurement in progress. The status light color corresponds to the selected light wavelength used for analysis.</td>
    </tr>
```

Which produces this:

<img width="229" height="132" alt="tasteTheRainbow" src="https://github.com/user-attachments/assets/010bec51-1af2-4cc7-8a3e-c4bf3c688ae8" />

### Detection wavelengths text

This short section lists the wavelengths the module uses for detection. I added color codes to the text in parenthesis (red, green, blue, etc.) to match the wavelengths. Thought that might be a neat touch. Can remove. See `specifications.md`, section starts on line 97.

## Review requests

Short manual. Shouldn't be much to review.

## Risk assessment

Low, docs change only.

jwwojak committed:
Template created

jwwojak committed:
First commit

jwwojak committed:
Mid morning save

jwwojak committed:
Safety section commit

jwwojak committed:
Specifications commit

Includes CSS to support a spectrum/rainbow status dot.

jwwojak committed:
Deck placement commit

jwwojak committed:
Installation commit

jwwojak committed:
Full text commit, no images

jwwojak committed:
Mid-morning save

jwwojak committed:
Final commit and then PR

jwwojak committed:
Merge branch 'edge' into plate-reader-manual

ecormany commented on docs/mkdocs.yml:
```diff
@@ -4,14 +4,15 @@ docs_dir: ./shared
-      - Thermocycler Module: '!include thermocycler-manual/mkdocs.yml'
-  - Protocol Designer: '!include protocol-designer/mkdocs.yml'
-  - Python API: https://docs.opentrons.com/v2/
+  - Opentrons Flex: '!include flex/mkdocs.yml'
```
No idea what's changed here, but it's completely broken the site nav. Revert to previous and re-insert the Absorbance Plate Reader line.

jwwojak replied:
Unicode `U+00A0` non-breaking page spaces

<img width="819" height="423" alt="non-breaking page spaces" src="https://github.com/user-attachments/assets/0684f7c3-dd86-4e6f-ba10-ea2a6001de54" />

jwwojak replied:
Fixed in recent push. No idea how or where these came from.

ecormany commented on docs/mkdocs.yml:
```diff
@@ -4,14 +4,15 @@ docs_dir: ./shared
+      - Flex Stacker: '!include stacker-manual/mkdocs.yml'
+      - HEPA/UV Module: '!include hepa-uv/mkdocs.yml'
+      - Thermocycler Module: '!include thermocycler-manual/mkdocs.yml'
+      - Plate Reader Module: '!include plate-reader/mkdocs.yml'
```
`Absorbance Plate Reader Module` always for full name. Just `plate reader` for subsequent mentions.

jwwojak replied:
Revised as indicated.

ecormany commented on docs/plate-reader/mkdocs.yml:
```diff
@@ -0,0 +1,52 @@
+site_name: plate-reader
```
`absorbance-plate-reader` here. and rename the directory to match.

jwwojak replied:
Fixed.

ecormany commented on docs/plate-reader/docs/specifications.md:
```diff
@@ -0,0 +1,150 @@
+        <p>At or above 450 nm:</p>
+            <ul>
+                <li>≤1% from 0.0-2.0 OD</li>
+                <li>≤1.5% from 2.0-3.0 OD</li>
```
en dashes 4eva
```suggestion
      <td>0–4.0 optical density (OD)</td>
    </tr>
    <tr>
      <th>Resolution</th>
      <td>0.001 OD</td>
    </tr>
    <tr>
      <th>Accuracy</th>
      <td>
        <p>The maximum deviation between the determined value and the true value.</p>
        <p>At 405 nm:</p>
            <ul>
                <li>≤1.5% + 0.010 OD from 0.0–2.0 OD</li>
                <li>≤3% + 0.010 OD from 2.0–3.0 OD</li>
            </ul>
        <p>At or above 450 nm:</p>
            <ul>
                <li>≤1% + 0.010 OD from 0.0–2.0 OD</li>
                <li>≤1.5% + 0.010 OD from 2.0–3.0 OD</li>
            </ul>
      </td>
    </tr>
    <tr>
      <th>Reproducibility</th>
      <td>
        <p>The maximum deviation between the determined values when the measurement is repeated directly.</p>
            <ul>
                <li>≤0.5% + 0.005 OD from 0.0–2.0 OD</li>
                <li>≤1% + 0.010 OD from 2.0–3.0 OD</li>
            </ul>
      </td>
    </tr>
    <tr>
      <th>Linearity</th>
      <td>
        <p>The maximum deviation between the true and the determined increase of the value.</p>
        <p>At 405 nm:</p>
            <ul>
                <li>≤1.5% from 0.0–2.0 OD</li>
                <li>≤3% from 2.0–3.0 OD</li>
            </ul>
        <p>At or above 450 nm:</p>
            <ul>
                <li>≤1% from 0.0–2.0 OD</li>
                <li>≤1.5% from 2.0–3.0 OD</li>
```

jwwojak replied:
Changed regular dash - to en dash –

ecormany commented on docs/plate-reader/docs/specifications.md:
```diff
@@ -0,0 +1,150 @@
+</table>
+
+## Detection Wavelengths
+<!-- Too much? I kinda like it but can remove the colors. -->
```
These are cool. You can keep them if they pass accessibility color contrast guidelines. My intuition is that green and orange don't.

jwwojak replied:
Removing the color. It's a bridge too far.

ecormany commented on docs/plate-reader/docs/specifications.md:
```diff
@@ -0,0 +1,150 @@
+  </tbody>
+</table>
+
+## Input and Output Connections
```
Different order than the PDF but it's fine here.

jwwojak replied:
I think this is a better location for it and removed an unnecessary header.

ecormany commented on docs/plate-reader/docs/specifications.md:
```diff
@@ -0,0 +1,150 @@
+
+The Absorbance Plate Reader has the following power input requirements, which are met by its USB connection to a Flex robot.
+
+### Power Supply (DC)
```
This looks funny in the nav as the lone H3. I think it's better as plain text.

jwwojak replied:
Removed H3

ecormany commented on docs/plate-reader/docs/data.md:
```diff
@@ -0,0 +1,14 @@
+
+# Using Plate Reader Data
+
+Plate reader data is available on the Recent Protocol Runs screen in the Opentrons App. To download your data from the app, click Devices and select your Flex from the list. Next, find the protocol that used the plate reader. Expand that protocol and click the Download link to retrieve your data. The robot formats plate reader data as a comma-separated value (.csv) file.
```
Bold the clickable things in this ¶

jwwojak replied:
Added bold to user-interactive UI features.

- **Devices**
- **Download**

ecormany commented on docs/plate-reader/docs/data.md:
```diff
@@ -0,0 +1,14 @@
+![CSV data file location](images/csv-data.png){width="90%"} <!-- 100% seems huge -->
+</figure>
+
+You can also use plate reader data immediately in a Python protocol. See the Python Protocol API documentation for more information.
```
Link to https://docs.opentrons.com/v2/modules/absorbance_plate_reader.html#using-plate-reader-data

jwwojak replied:
Added link.

ecormany commented on docs/plate-reader/mkdocs.yml:
```diff
@@ -0,0 +1,52 @@
+  - Deck Placement: placement.md
+  - Installation: installation.md
+  - Using Plate Reader Data: data.md
+  - Additional Product Information: warranty.md
```
Maintenance page missing? No .md file for it either.

jwwojak replied:
Yikes. That's bad. 📃 Adding.

ecormany reviewed (changes requested):
Handful of updates required. Most important is fixing the mysterious change to the top-level `mkdocs.yml` file that breaks other publications.

jwwojak committed:
Lunch commit to see what breaks

jwwojak committed:
More review changes

- bold UI elements
- add missing maintenance doc
- remove dashes for en dashes
- update yml file

jwwojak committed:
change to match other manuals

ecormany commented on docs/shared/modules/index.md:
```diff
@@ -38,4 +38,12 @@ Opentrons hardware modules add features and capabilities to your robot.
+
+    A spectrophotometer that uses light absorbance to determine sample concentrations.
+
+    [:octicons-arrow-right-24: Instruction Manual](../plate-reader/index.md)
```
Update link.
```suggestion
    [:octicons-arrow-right-24: Instruction Manual](../absorbance-plate-reader/index.md)
```

jwwojak replied:
Oh! that's a subtle one! Committing it now.

jwwojak committed:
More reviews before other reviews

- remove opentrons logo from cover page, module title onl.
- remove punctuation from company name, the "Inc."
- fixed a leading space in a link in data.md
- minor text changes here and there
- LED table spacing

jwwojak committed:
Update docs/shared/modules/index.md

Co-authored-by: Ed Cormany <edward.cormany@opentrons.com>

jwwojak committed:
Merge branch 'edge' into plate-reader-manual

jwwojak commented:
Fixed conflicts.

jwwojak committed:
Remove H1 headers

Let the nav control the page title.

jwwojak committed:
I can haz wordz

Fix spelling typo in title.

ecormany commented on docs/shared/modules/index.md:
```diff
@@ -46,4 +46,12 @@ Opentrons hardware modules add features and capabilities to your robot.
 
     [:octicons-arrow-right-24: Instruction Manual](../thermocycler-manual/index.md)
 
+-   :opentrons-platereaderout:{ .xl .middle} __Plate Reader__
```
Full product name
```suggestion
-   :opentrons-platereaderout:{ .xl .middle} __Absorbance Plate Reader__
```

jwwojak replied:
I see it. So as to get into the habit, I'm going to make the change locally and then commit the changes. This will help me remember to modify the other shared files, when they're needed.

ecormany commented on docs/absorbance-plate-reader/docs/data.md:
```diff
@@ -0,0 +1,12 @@
+![CSV data file location](images/csv-data.png){width="90%"} <!-- 100% seems huge -->
+</figure>
+
+You can also use plate reader data immediately in a Python protocol. See the [Python Protocol API](https://docs.opentrons.com/v2/modules/absorbance_plate_reader.html#using-plate-reader-data) for more information.
```
Missing word
```suggestion
You can also use plate reader data immediately in a Python protocol. See the [Python Protocol API documentation](https://docs.opentrons.com/v2/modules/absorbance_plate_reader.html#using-plate-reader-data) for more information.
```

jwwojak replied:
Will fix this also with the changes to `modules/index.md`

ecormany reviewed (approved):
Literally two words to add and then this is good to merge!

jwwojak committed:
2 final changes

- Fix name in `docs/shared/modules/index.md` to include "Absorbance" in the name.

- In Plate Reader `data.md` file, add "documentation" to the link to the API docs.
